### PR TITLE
Added code analyzer

### PR DESCRIPTION
**Pull request by mohakgupt**
Fixes appwrite/appwrite#4389.
I have used PHPStan as the analyzer. Also fixed the errors thrown by the analyzer.

Screenshot 2022-10-21 020220

### CODE REVIEW

**Comment by mohakgupt** [comment]
@Meldiron any update on this

**Comment by PineappleIOnic** [comment]
Thank you so much for the PR 🤩. We're adding the `hacktoberfest-accepted` label to ensure this PR counts towards your Hacktoberfest contributions count. With that said, please stay active on this PR to address any comments once you receive a review. Happy Hacktoberfest! 🎃

**Comment by stnguyen90** [comment]
I still got an error when I ran the check:

image

**Comment by mohakgupt** [comment]
> I still got an error when I ran the check:
> 
> image

It is running the test on max level. Running `./vendor/bin/phpstan analyse --level 8 src tests` will give no errors. I have fixed the errors on level 8 as in the issue description it was said to use level 8. Thanks.

**Review by stnguyen90 — DISMISSED** [review]
Great PR! 🤯 Would you please fix the merge conflicts and review the comments I left.

File: src/Audit/Audit.php [hunk 1]
```diff
@@ -183,9 +183,9 @@ public function log(string $userId, string $event, string $resource, string $use
...
      * @param Document|null $orderAfter
      *
-     * @return array
+     * @return mixed
      */
-    public function getLogsByUser(string $userId, int $limit = 25, int $offset = 0, Document $orderAfter = null): array
+    public function getLogsByUser(string $userId, int $limit = 25, int $offset = 0, Document $orderAfter = null): mixed
     {
         $result = Authorization::skip(function () use ($userId, $limit, $offset, $orderAfter) {
             $queries = $this->buildQuery(['userId' => $userId], Query::TYPE_EQUAL);
```
Comment (stnguyen90) [inline src/Audit/Audit.php:191]: The typing for `Authorization::skip()` makes this a little weird, but we can do this for now:

```suggestion
    public function getLogsByUser(string $userId, int $limit = 25, int $offset = 0, Document $orderAfter = null): array
    {
        /** @var Document[] $result */
        $result = Authorization::skip(function () use ($userId, $limit, $offset, $orderAfter) {
            $queries = $this->buildQuery(['userId' => $userId], Query::TYPE_EQUAL);
```

**Comment by mohakgupt** [comment]
I have done all the changes.

**Review by christyjacob4 — APPROVED** [review]

**Comment by christyjacob4** [comment]
THANK YOU! All changes merged 🥳

Please reach out to me on our [Discord server](https://appwrite.io/discord) if you would like to claim your Appwrite swags! As a way of saying thank you, we would also love to invite you to join the Appwrite organization on GitHub. Please share your GitHub username with us on Discord.  

You can accept the invite by visiting https://github.com/orgs/appwrite/invitation. By joining our team, you will officially be an Appwrite maintainer on GitHub.

You can change your membership [visibility settings](https://help.github.com/en/github/setting-up-and-managing-your-github-user-account/publicizing-or-hiding-organization-membership), so your new Appwrite team membership badge will show up on your personal GitHub profile.

Please feel free to look for more PRs you might be interested in helping with on our long list of [Hacktoberfest friendly issues](https://github.com/search?q=org%3Aappwrite+org%3Autopia-php+org%3Aopen-runtimes+label%3Ahacktoberfest+created%3A%3E2021-01-01+state%3Aopen&type=issues&s=updated&o=asc) and help make Appwrite better :)

**Pull request merged**